wagtail/search/backends/database/mysql/query.py

```diff
@@ -0,0 +1,208 @@
+        self.extra = extra
+        if isinstance(value, str) or search_type == 'phrase':  # If the value is a string, we assume it's a phrase
+            safe_string = re.sub(r'\W+', ' ', value)  # Remove non-word characters. This is done to disallow the usage of full text search operators in the MATCH clause, because MySQL doesn't include these kinds of characters in FULLTEXT indexes.
+            self.value = Value('"%s"' % safe_string)  # We wrap it in quotes to make sure it's parsed as a phrase
+        else:  # Otherwise, we assume it's a lexeme
+            self.value = value
+
+    def as_sql(self, compiler: SQLCompiler, connection: BaseDatabaseWrapper, **extra_context: Any) -> Tuple[str, List[Any]]:
+        sql, params = compiler.compile(self.value)
+        return (sql, params)
+
+    def __repr__(self) -> str:
+        return self.value.__repr__()
+
+
+class CombinedSearchQuery(SearchQueryCombinable, CombinedExpression):
+    def __init__(self, lhs, connector, rhs, output_field=None):
+        super().__init__(lhs, connector, rhs, output_field)
+
+    def __str__(self):
+        return '%s' % super().__str__()
+
+    def as_sql(self, compiler, connection):
+        value_params = []
+
+        lhs_connector = self.connector
+        rhs_connector = self.connector
+
+        if isinstance(self.lhs, SearchQuery) and isinstance(self.lhs.value, Lexeme) and self.lhs.value.invert and self.connector == '+':  # NOTE: The explanation for this special case is the same as above, in the CombinedLexeme class.
+            modified_value = self.lhs.value.copy()
+            modified_value.invert = not modified_value.invert
+            lhs_connector = '-'
+            lsql, params = compiler.compile(modified_value)
+        else:
+            lsql, params = compiler.compile(self.lhs)
+        value_params.extend(params)
+
+        if isinstance(self.rhs, SearchQuery) and isinstance(self.rhs.value, Lexeme) and self.rhs.value.invert and self.connector == '+':  # NOTE: The explanation for this special case is the same as above, in the CombinedLexeme class.
+            modified_value = self.rhs.value.copy()
+            modified_value.invert = not modified_value.invert
+            rhs_connector = '-'
+            rsql, params = compiler.compile(modified_value)
+        else:
+            rsql, params = compiler.compile(self.rhs)
+        value_params.extend(params)
+
+        combined_sql = '({}{} {}{})'.format(lhs_connector, lsql, rhs_connector, rsql)  # if self.connector is '+' (AND), then both terms will be ANDed together. We need to repeat the connector to make that work.
+        combined_value = combined_sql % tuple(value_params)
+        return '%s', [combined_value]
+
+
+class MatchExpression(Expression):
+    filterable = True
+    template = 'MATCH (%s) AGAINST (%s IN BOOLEAN MODE)'
+
+    def __init__(self, query: SearchQueryCombinable, columns: List[str] = ['title', 'body'], output_field: Field = BooleanField()) -> None:
```

Comment:
The best practice in Python is to not use lists as default parameters since they are initialised only once and reused (so if it's modified in any way, the next call will get the modified list and not a fresh one).